MF-in-5-min.md

```diff
@@ -0,0 +1,178 @@
+
+Fixes can be provided inline as text argument of the fix module in the Flux script, or as a pointer to a Fix Script. A Fix Scripts groups one or more fixes in a file.
+
+```
+$ cat example.fix
+add_field('address.street','Walker Street')
+add_field('address.number','15')
+copy_field('colors.2','best_color')
+```
+
+```
+$ cat data.yaml
+---
+colors:
+- Red
+- Green
+- Blue
+...
+```
+
+```
+$ cat yaml2yaml.flux
+"data.yaml"
+| open-file
+| as-records
+| decode-yaml
+| fix("example.fix")
+| encode-yaml
+| print
+;
+```
+
+```
+---
+address:
+    number: '15'
+    street: Walker Street
+best_color: Green
+colors:
+    - Red
+    - Green
+    - Blue
+...
+```
+[Example in Playground](https://metafacture.org/playground/?flux=inputFile%0A%7C+open-file%0A%7C+as-records%0A%7C+decode-yaml%0A%7C+fix%28transformationFile%29%0A%7C+encode-yaml%0A%7C+print%0A%3B&transformation=add_field%28%27address.street%27%2C%27Walker+Street%27%29%0Aadd_field%28%27address.number%27%2C%2715%27%29%0Acopy_field%28%27colors.2%27%2C%27best_color%27%29&data=---%0Acolors%3A%0A-+Red%0A-+Green%0A-+Blue)
+
+In the example we created the Fix Script `example.fix` that contains a combination of mappings and data conversion on (nested) data. We run a YAML to YAML conversion using the example.fix Fix Script. (Note: Differently to Catmandu array/list index are 1-based.)
+
+
+## SPECIALIZATIONS
+
+Metafacture was mainly created for data conversions of specialized metadata languages in the field of libraries, archives and museums. One of the specialized modules is `decode-marc` and `encode-marc`. 
+
+With them you can read, write and convert MARC files.
+
+For instance, to extract all the titles from an ISO MARC file one could write:
+
+```
+$ cat titles.fix
+set_array("title")
+copy_field("245??.?","title.$append")
+join_field("title")
+copy_field("001","id")
+retain("title", "id")
+```
+ 
+ ```
+ $ cat marc2csv.flux
+"data.mrc"
+| open-file
+| as-records
+| decode-marc21
+| fix("title.fix")
+| encode-csv
+| print
+;
+```
+
+[Example in Playground](https://metafacture.org/playground/?flux=%22https%3A//raw.githubusercontent.com/metafacture/metafacture-core/master/metafacture-runner/src/main/dist/examples/read/marc21/10.marc21%22%0A%7C+open-http%0A%7C+as-lines%0A%7C+decode-marc21%0A%7C+fix%28transformationFile%29%0A%7C+encode-csv%0A%7C+print%0A%3B&transformation=set_array%28%22title%22%29%0Acopy_field%28%22245%3F%3F.%3F%22%2C%22title.%24append%22%29%0Ajoin_field%28%22title%22%29%0Acopy_field%28%22001%22%2C%22id%22%29%0Aretain%28%22title%22%2C+%22id%22%29)
+
+In the example above marc data is converted to a csv file.
+The 245 field with its subfields of each MARC record is mapped to the title field.
+The retain Fix function keeps only the title field in the output. (In contrast to Catmandu there are no special marc or pica fixes since the iternal handling of records and elements is more generic. Also the internal serialization of MARC is not as complex as in Catmandu.)
```

Comment:
```suggestion
The `retain` Fix function keeps only the title field in the output. (In contrast to Catmandu there are no special marc or pica fixes since the internal handling of records and elements is more generic. Also the internal serialization of MARC is not as complex as in Catmandu.)
```